src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/CCDCaseApi.java

```diff
@@ -65,45 +76,125 @@ public CCDCaseApi(
     }
 
     public Optional<Claim> getByExternalId(String externalId, String authorisation) {
-        final List<Claim> claims = search(authorisation, ImmutableMap.of("case.externalId", externalId));
+        User user = userService.getUser(authorisation);
+        return getCaseDetailsByExternalId(user, externalId)
+            .map(CaseDetails::getData)
+            .map(this::convertToCCDCase)
+            .map((caseMapper::from));
+    }
 
-        if (claims.size() > 1) {
+    public Optional<Claim> linkDefendant(String externalId, String defendantId, String authorisation) {
+        User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+        Optional<CaseDetails> optionalCaseDetails = getCaseDetailsByExternalId(anonymousCaseWorker, externalId);
+
+        if (optionalCaseDetails.isPresent()) {
+            CaseDetails caseDetails = optionalCaseDetails.get();
+            caseAccessApi.grantAccessToCase(anonymousCaseWorker.getAuthorisation(),
+                authTokenGenerator.generate(),
+                anonymousCaseWorker.getUserDetails().getId(),
+                JURISDICTION_ID,
+                CASE_TYPE_ID,
+                caseDetails.getId().toString(),
+                new UserId(defendantId)
+            );
+
+            User defendant = userService.getUser(authorisation);
+            return Optional.of(readCase(defendant, caseDetails.getId().toString()));
+        }
+        return Optional.empty();
+    }
+
+    public List<Claim> getByDefendantId(String id, String authorisation) {
+        User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+        List<String> caseIdsGivenUserIdHasAccessTo = caseAccessApi.findCaseIdsGivenUserIdHasAccessTo(
+            anonymousCaseWorker.getAuthorisation(),
+            authTokenGenerator.generate(),
+            anonymousCaseWorker.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            id
+        );
+
+        User defendant = userService.getUser(authorisation);
+        return caseIdsGivenUserIdHasAccessTo.stream()
+            .map((caseId) -> readCase(defendant, caseId))
+            .filter((claim -> !claim.getSubmitterId().equals(id)))
+            .collect(Collectors.toList());
+    }
+
+    public Optional<Claim> getByLetterHolderId(String id, String authorisation) {
+        User anonymousCaseWorker = userService.authenticateAnonymousCaseWorker();
+
+        List<String> letterHolderCases = caseAccessApi.findCaseIdsGivenUserIdHasAccessTo(
+            anonymousCaseWorker.getAuthorisation(),
+            authTokenGenerator.generate(),
+            anonymousCaseWorker.getUserDetails().getId(),
+            JURISDICTION_ID,
+            CASE_TYPE_ID,
+            id
+        );
+        if (letterHolderCases.size() > 1) {
+            throw new DefendantLinkingException("More than 1 case a letter holder ID has access to found");
+        }
+        if (letterHolderCases.size() == 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.reform.hmcts.net/coding_rules#rule_key=squid%3AS1155)